twine/settings.py

```diff
@@ -222,6 +224,15 @@ def register_argparse_arguments(parser: argparse.ArgumentParser) -> None:
             "(package index) with. (Can also be set via "
             "%(env)s environment variable.)",
         )
+        parser.add_argument(
+            "--trusted-publish",
+            default=False,
+            required=False,
+            action="store_true",
+            help="Upload from CI using trusted publishing. Use this without "
+            "specifying username & password. Requires an optional extra "
```

Comment:
Most folks aren't using username/password today. More of an API token for PyPI. Maybe it would be better to say that it avoids needing to manage credentials for the CI system?